avocado/core/nrunner.py

```diff
@@ -10,6 +10,9 @@
 import unittest
 import socket
 
+from .tapparser import TapParser
```

Comment:
The dependency on other Avocado modules is something that we'll eventually need to tackle properly.  I've been trying to keep `avocado/core/nrunner.py` free from Avocado dependencies because it would allow a easy deployment into containers, vms, initrd images, etc.  That's why support for `avocado-instrumented` was added in `avocado/core/nrunner_avocado_instrumented.py`.

But, we should keep evaluating what will be costlier: separating Python based runners that could or need to use the Avocado APIs, or having a more complex deployment model (or none at all, leaving it up to users).

Reply:
I got your point.

Unless I'm missing something, I just have a feeling (so far) that maintain code is harder than deployment. We will probably have duplicated code, as I can see.

IMO, have a clean and separated code is easy to get more contributors and to maintain code.

Today, we provide few options to the user deployment:

 * git clone;
 * pip;
 * rpm repos;
 * manual (ssh/rsync over the entire package);

I understand that those methods are not "so easy" than a scp/rsync of only one file and maybe I will feel the pain when I feel the problem on my skin.